src/interop/js-interop/usage.md

```diff
@@ -0,0 +1,368 @@
+---
+title: Usage
+description: How to declare and use JS interop members.
+---
+
+JS interop works by looking up properties in the [global JS scope]. In order to
+interact with these properties, you use [`external`] interop members. In order
+to provide types for these properties, you use interop types.
+
+## Interop types
+
+When interacting with a JS value, you need to provide a Dart type for it. In
+order to do this, you either use or declare an interop type. Dart provides a set
+of core interop types called ["JS types"] that you can use to add types to your
+interop members. You can also declare an interop type using [extension types] by
+making its representation type another interop type. "JS types" refer to the
+interop types provided by Dart, and while all JS types are interop types, not
+all interop types are JS types. For example:
```

Comment:
Consider a few simplifications for this paragraph:
- drop the last clarification that not all interop types are js types
- use a recursive definition (can help make the above simplification more obvious)
- split what interop types are used for to a separate paragraph. The reference above that they are used "to add types to interop members" can be confusing because we use the word "types" 3 times in that sentence.

All in all, a suggestion for this paragraph could be:
> Interop types let you provide a Dart type for interacting with a JS value. Interop types are either a core [JS type] provided by Dart, or an [extension type] wrapping an existing interop type.

Then later we can say how they are used, for example:

> Interop types allow you to provide an interface for how members of a JS values can be used. They are also used in the signature of other interop APIs to declare assumptions about inputs and outputs.